homeassistant/components/notify/pushsafer.py

```diff
@@ -27,21 +56,125 @@
+                picture1_encoded = self.loadfromurl(url, username, password,
+                                                    auth)
+            elif local_path is not None:
+                _LOGGER.debug("Loading image from file %s", local_path)
+                picture1_encoded = self.loadfromfile(local_path)
+            else:
+                _LOGGER.warning("missing url or local_path for picture1")
+        else:
+            _LOGGER.debug("picture1 is not specified")
+
+        payload = {
+            'k': self._private_key,
+            't': title,
+            'm': message,
+            's': data.get(ATTR_SOUND, ATTR_SOUND_DEFAULT),
+            'v': data.get(ATTR_VIBRATION, ATTR_VIBRATION_DEFAULT),
+            'i': data.get(ATTR_ICON, ATTR_ICON_DEFAULT),
+            'c': data.get(ATTR_ICONCOLOR, ATTR_ICONCOLOR_DEFAULT),
+            'u': data.get(ATTR_URL, ATTR_URL_DEFAULT),
+            'ut': data.get(ATTR_URLTITLE, ATTR_URLTITLE_DEFAULT),
+            'l': data.get(ATTR_TIME2LIVE, ATTR_TIME2LIVE_DEFAULT),
+            'p': picture1_encoded
+        }
+
+        _LOGGER.debug("using push data: %s", str(payload))
+
+        for target in targets:
+            payload['d'] = target
+            response = requests.post(_RESOURCE, data=payload,
+                                     timeout=CONF_TIMEOUT)
+            if response.status_code != 200:
+                _LOGGER.error("Pushsafer failed with: %s", response.text)
+            else:
+                _LOGGER.debug("Push send: %s", response.json())
+
+    @classmethod
+    def getbase64(cls, filebyte, mimetype):
+        """Convert the image to the expected base64 string of pushsafer."""
+        if mimetype not in _ALLOWED_IMAGES:
+            _LOGGER.warning("%s is a not supported mimetype for images",
+                            mimetype)
+            return None
+        else:
+            if filebyte is not None:
+                base64_image = base64.b64encode(filebyte).decode('utf8')
+                return "data:" + mimetype + ";base64," + base64_image
+            else:
+                _LOGGER.warning("Base64 encode no image passed")
+
+    def loadfromurl(self, url=None, username=None, password=None, auth=None):
+        """Load image/document/etc from URL."""
+        if url is not None:
+            _LOGGER.debug("Downloading image from %s", url)
+            if username is not None and password is not None:
+                auth_ = HTTPBasicAuth(username, password)
+                response = requests.get(url, auth=auth_,
+                                        timeout=CONF_TIMEOUT)
+            else:
+                response = requests.get(url, timeout=CONF_TIMEOUT)
+            return self.getbase64(response.content,
+                                  response.headers['content-type'])
+        else:
+            _LOGGER.warning("url not found in param")
+
+        return None
+
+    def loadfromfile(self, local_path=None):
```

Comment:
See above.

Reply:
made your suggested changes

Reply:
Please update this method name to be snakecase.

Reply:
done